resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala

```diff
@@ -134,6 +166,42 @@ class YarnAllocatorSuite extends SparkFunSuite with Matchers with BeforeAndAfter
     size should be (0)
   }
 
+  test("custom resource type requested from yarn") {
+    assume(ResourceRequestHelper.isYarnResourceTypesAvailable())
+    TestYarnResourceRequestHelper.initializeResourceTypes(List("gpu"))
+
+    // request a single container and receive it
+    val handler = createAllocatorWithAdditionalConfigs(1, Map(
+      YARN_EXECUTOR_RESOURCE_TYPES_PREFIX + "gpu" -> "2G",
+      YARN_EXECUTOR_RESOURCE_TYPES_PREFIX + "memory" -> "1G"
+    ))
+    handler.updateResourceRequests()
+    handler.getNumExecutorsRunning should be (0)
+    handler.getPendingAllocate.size should be (1)
+
+    val resource = Resource.newInstance(3072, 6)
+    val resourceTypes = Map("gpu" -> "2G")
+    ResourceRequestHelper.setResourceRequests(resourceTypes, resource)
+
+    val container = createContainerWithResource("host1", resource)
+    handler.handleAllocatedContainers(Array(container))
+
+    // verify custom resource type is part of rmClient.ask set
+    val askField = rmClient.getClass.getDeclaredField("ask")
```

Comment:
I'm not sure this is testing anything interesting. It's basically testing YARN code, not Spark code, and using reflection for that...

The code you're actually adding to `YarnAllocator` is not being tested, which is the code that creates the `resource` field in that class. Instead, you're manually creating a `Resource` in this test.

Testing that code would be better than what you have here.

Reply:
Will look into this tomorrow morning

Reply:
Good point.
I found a better way to test what I wanted and it also covers verifying whether the `resource` field is set correctly.
Please check the new testcase's implementation.